src/firebase/docs/user.js

```diff
@@ -0,0 +1,22 @@
+import { collection, addDoc } from 'firebase/firestore'
+import { db } from '../index'
+
+// Function to register a user and save data in Firestore
+export const registerUserData = async (uid, name, email, age, country) => {
+   try {
+      // Get a reference to the "users" collection
+      const usersCollection = collection(db, 'users')
+      // Add a new document to the "users" collection
+      const docRef = await addDoc(usersCollection, {
```

Comment:
There is a way to sync the id, so that it match with the user id in authentication, addDoc creates an id for us to keep track of the doc, because of that we are most likely to use setDoc which we can specify the doc to be updated, we can specify the doc with the id provided in the credentials response from createUserWithEmailAndPassword